### DISCUSSION TITLE
Additional tag pushed to registry during signature

### DISCUSSION THREAD

**Discussion by dalarson** [notaryproject/notation#979] [Q&A] [answered]
Hello,

I am attempting to add a signature step to my build pipeline using notation. The docker image gets build and tagged with the semantic version of the code repository, so in the container registry we have a series of tags that map directly to commits in the git repo. The notation cli step that I've added to the build pipeline seems to sign images correctly, but for some reason it also seems to be pushing another tag to the container registry. The result is that the image initially built and tagged is signed, but there's another copy of the same signed image with the digest as the tag as shown here:
![Screenshot 2024-06-28 at 10 32 01 AM](https://github.com/notaryproject/notation/assets/23143175/a76f661a-1da2-4869-a0a4-4f739698b9a1)
My question is how can I prevent the additional tag from being pushed? The behavior I would like would be that the existing image tag gets signed, and no additional tags are pushed to the registry.

Guidance would be greatly appreciated

**Comment by FeynmanZhou** [comment]
Hi @dalarson , 

This is because notation uses the [tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) to store the signature in registry. 

The signature is configurable. You can choose to sign an image with [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if your registry is compliant with OCI v1.1.

May I know which registry you are using and your notation version?

**Reply by dalarson** [reply]
Hello @FeynmanZhou! Thank you for your response. I am using Azure Container Registry and notation v 1.1.0. I believe the registry should be OCI compliant. I need to read a bit about the referrers API to understand what it's all about.

**Reply by FeynmanZhou** [reply, marked as answer]
@dalarson 
Ah, ACR is an OCI v1.1 compliant registry so you can switch to use referrers API  to store the signature. Learn how OCI v1.1 and referrers API benefits ACR: https://techcommunity.microsoft.com/t5/apps-on-azure-blog/announcing-support-of-oci-v1-1-specification-in-azure-container/ba-p/4177906

This feature is still an [experimental feature](https://notaryproject.dev/docs/user-guides/experimental-features/) in Notation v1.1.x, you can resign images using referrers API by following these steps:

1. To enable experimental features, set the NOTATION_EXPERIMENTAL environment variable to 1,

```
export NOTATION_EXPERIMENTAL=1
```
2. Sign an image using referrers API 

```
notation sign --allow-referrers-api <registry>/<repository>@<digest>
```

**Reply by FeynmanZhou** [reply]
Hi @dalarson , does the guidance above work for you?

**Reply by dalarson** [reply]
Thank you for the information. Yes this is working beautifully for me. :)

Cheers!

**Reply by FeynmanZhou** [reply]
Cool. Thanks for confirming it! @dalarson

**Comment by TnGadaria** [comment]
Hi @dalarson,
Alternatively, you can use this: https://github.com/notaryproject/notation/tree/v1.2.0-alpha.1
